nemo/collections/asr/models/ctc_models.py

```diff
@@ -663,6 +705,72 @@ def test_dataloader(self):
         if self._test_dl is not None:
             return self._test_dl
 
+    def _setup_pseudo_label_dataloader(self, cache_manifest: str, audio_tar: str = None, batch_size: int = 64):
+
+        if self.cfg.train_ds.get("is_tarred", False):
+
+            dl_config = {
+                'manifest_filepath': cache_manifest,
+                'tarred_audio_filepaths': audio_tar,
+                'sample_rate': self.preprocessor._sample_rate,
+                'labels': OmegaConf.to_container(self.decoder.vocabulary),
+                'is_tarred': True,
+                'use_lhotse': True,
+                'shard_manifests': False,
+                'tarred_shard_strategy': 'replicate',
+                'batch_size': batch_size,
+                'drop_last': False,
+                'trim_silence': False,
+                'shuffle': False,
+                'shuffle_n': 0,
+                'num_workers': self.cfg.train_ds.num_workers,
+                'pin_memory': True,
+                'tarred_random_access': True,
+            }
+
+            dl_config = OmegaConf.create(dl_config)
+
+            return get_lhotse_dataloader_from_config(
+                dl_config,
+                global_rank=self.global_rank,
+                world_size=self.world_size,
+                dataset=LhotseSpeechToTextBpeDataset(
```

Comment:
is it sure bpe dataset here?

Reply:
Yes, we parse the tokenizer accordingly.